src/main/java/net/dries007/tfc/common/fluids/MixingFluid.java

```diff
@@ -61,6 +62,13 @@ public boolean isSourceBlockOfThisType(FluidState stateIn)
         return stateIn.getType().isSame(this) && stateIn.isSource();
     }
 
+    @Override
+    public boolean isSame(Fluid fluid) {
+        return super.isSame(fluid) || fluid == TFCFluids.RIVER_WATER.get()
+            || fluid == TFCFluids.SALT_WATER.getSource() || fluid == TFCFluids.SALT_WATER.getFlowing()
+            || fluid == Fluids.WATER || fluid == Fluids.FLOWING_WATER;
+    }
```

Comment:
Is there a better way we can express this? Like with a tag the fluid properties or something?

Reply:
Yeah, I think the ANY_INFINITE_WATER tag has what we need for this, though we could create another tag with the same entries if you think that would be cleaner